src/pybind/mgr/rbd_support/module.py

```diff
@@ -483,13 +501,733 @@ def get_perf_counters(self, pool_spec, sort_by):
+        except rbd.ImageNotFound:
+            task.fail("Image does not exist")
+            self.log.info("{}: task={}".format(task.failure_message, str(task)))
+
+    def execute_migration_execute(self, ioctx, task):
+        self.log.info("execute_migration_execute: task={}".format(str(task)))
+
+        try:
+            rbd.RBD().migration_execute(ioctx, task.refs[TASK_REF_IMAGE_NAME],
+                                        on_progress=partial(self.progress_callback, task))
+        except rbd.ImageNotFound:
+            task.fail("Image does not exist")
+            self.log.info("{}: task={}".format(task.failure_message, str(task)))
+        except rbd.InvalidArgument:
+            task.fail("Image is not migrating")
+            self.log.info("{}: task={}".format(task.failure_message, str(task)))
+
+    def execute_migration_commit(self, ioctx, task):
+        self.log.info("execute_migration_commit: task={}".format(str(task)))
+
+        try:
+            rbd.RBD().migration_commit(ioctx, task.refs[TASK_REF_IMAGE_NAME],
+                                       on_progress=partial(self.progress_callback, task))
+        except rbd.ImageNotFound:
+            task.fail("Image does not exist")
+            self.log.info("{}: task={}".format(task.failure_message, str(task)))
+        except rbd.InvalidArgument:
+            task.fail("Image is not migrating or migration not executed")
+            self.log.info("{}: task={}".format(task.failure_message, str(task)))
+
+    def execute_migration_abort(self, ioctx, task):
+        self.log.info("execute_migration_abort: task={}".format(str(task)))
+
+        try:
+            rbd.RBD().migration_abort(ioctx, task.refs[TASK_REF_IMAGE_NAME],
+                                      on_progress=partial(self.progress_callback, task))
+        except rbd.ImageNotFound:
+            task.fail("Image does not exist")
+            self.log.info("{}: task={}".format(task.failure_message, str(task)))
+        except rbd.InvalidArgument:
+            task.fail("Image is not migrating")
+            self.log.info("{}: task={}".format(task.failure_message, str(task)))
+
+    def complete_progress(self, task):
+        self.log.debug("complete_progress: task={}".format(str(task)))
+        try:
+            if task.failed:
+                self.module.remote("progress", "fail", task.task_id,
+                                   task.failure_message)
+            else:
+                self.module.remote("progress", "complete", task.task_id)
+        except ImportError:
+            # progress module is disabled
+            pass
+
+    def update_progress(self, task, progress):
+        self.log.debug("update_progress: task={}, progress={}".format(str(task), progress))
+        try:
+            self.module.remote("progress", "update", task.task_id,
+                               task.message, progress,
+                               {**{"origin": "rbd_support"}, **task.refs})
```

Comment:
Do we still support python2?

I happened to try this on Ubuntu 16.04, built with python2.7, and observing this error:
```
SyntaxError: ('invalid syntax', ('/home/mgolub/ceph/ceph.ci/src/pybind/mgr/rbd_support/module.py', 923, 34, '                               {**{"origin": "rbd_support"}, **task.refs})\n'))
```
which I believe due to python2.7?

Reply:
Switch this to a `{}.update()` call